domain/model/service/service.go

```diff
@@ -242,12 +263,38 @@ func (s *Service) ModelType(ctx context.Context, uuid coremodel.UUID) (coremodel
 func (s *Service) DeleteModel(
 	ctx context.Context,
 	uuid coremodel.UUID,
+	opts ...model.DeleteModelOption,
 ) error {
+	options := model.DefaultDeleteModelOptions()
+	for _, fn := range opts {
+		fn(options)
+	}
+
 	if err := uuid.Validate(); err != nil {
 		return fmt.Errorf("delete model, uuid: %w", err)
 	}
 
-	return s.st.Delete(ctx, uuid)
+	// Delete common items from the model. This helps to ensure that the
+	// model is cleaned up correctly.
+	if err := s.st.Delete(ctx, uuid); err != nil {
+		return fmt.Errorf("delete model: %w", err)
+	}
+
+	// If the db should not be deleted then we can return early.
+	if !options.DeleteDB() {
+		s.logger.Infof("skipping model deletion, model database will still be present")
```

Comment:
Do we really need logging in here? Surely this would be better from the callers side and also updating the comment of the function to explain expected default behaviour.

My reason for asking is I am really not a fan of logging in the services layer. The services layer has nice well defined behaviour that is documented and strong errors. Logging can get pushed up a layer with all of that done I think.

Reply:
So this only exists because we can't delete a model outside of model migration in its current state. The key part we're missing is going from `alive -> dying -> dead`. We should delete these options when models are correctly progressing through the life states.